semanticdb-gradle-plugin/src/main/scala/SemanticdbGradlePlugin.scala

```diff
@@ -0,0 +1,402 @@
+package com.sourcegraph.gradle.semanticdb
+
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.{util => ju}
+
+import scala.jdk.CollectionConverters._
+
+import com.sourcegraph.scip_java.BuildInfo
+import org.gradle.api.DefaultTask
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.artifacts.component.ModuleComponentIdentifier
+import org.gradle.api.tasks.TaskAction
+import org.gradle.api.tasks.compile.JavaCompile
+import org.gradle.api.tasks.scala.ScalaCompile
+
+class SemanticdbGradlePlugin extends Plugin[Project] {
+  override def apply(project: Project) = {
+    val gradle = new GradleVersion(project.getGradle().getGradleVersion())
+    project.afterEvaluate { project =>
+      project.getRepositories().add(project.getRepositories().mavenCentral())
+      project.getRepositories().add(project.getRepositories().mavenLocal())
+
+      val extra = project.getExtensions().getExtraProperties()
+      val extraProperties = extra.getProperties().asScala
+
+      val targetRoot = extra
+        .getProperties()
+        .asScala
+        .getOrElse("semanticdbTarget", project.getBuildDir())
+
+      val javacPluginVersion = BuildInfo.version
+
+      val javacPluginJar = extraProperties
+        .get("javacPluginJar")
+        .map(_.asInstanceOf[String])
+
+      val javacDep = javacPluginJar
+        .map[Object](jar => project.files(jar))
+        .getOrElse(s"com.sourcegraph:semanticdb-javac:${javacPluginVersion}")
+
+      val sourceRoot = project.getRootDir()
+      val agentJar = extraProperties.get("javacAgentPath").map(_.toString)
+
+      val tasks = project.getTasks()
+
+      val triggers = List.newBuilder[String]
+
+      if (project.getPlugins().hasPlugin("java")) {
+
+        triggers += "compileJava"
+        triggers += "compileTestJava"
+
+        val compilerPluginAdded =
+          try {
+            project.getDependencies().add("compileOnly", javacDep)
+            project.getDependencies().add("testCompileOnly", javacDep)
+            true
+          } catch {
+            case exc: Exception =>
+              System
+                .err
+                .println(
+                  s"Failed to add compiler plugin to javac, will go through the agent route: ${exc.getMessage()}"
+                )
+              false
+          }
+
+        project
+          .getTasks()
+          .withType(classOf[JavaCompile])
+          .configureEach { task =>
+            if (gradle.is5 || (gradle.is6 && !gradle.is6_7_plus))
+              println(
+                task
+                  .asInstanceOf[{
+                      def getToolChain(): Any
+                    }
+                  ]
+                  .getToolChain()
+              )
```

Comment:
TODO: see what we can do about this.

Reply:
Given Gradle's compatibility matrix (only 7.3+ supports Java 17) this branch doesn't need anything.

I have a spearate PR coming that will clean this up among other things